tests/core/pyspec/eth2spec/utils/bls.py

```diff
@@ -121,24 +171,123 @@ def AggregatePKs(pubkeys):
         # milagro_bls._AggregatePKs checks KeyValidate internally
         pass
 
+    if bls == arkworks_bls:  # no signature API in arkworks
+        return py_ecc_bls._AggregatePKs(list(pubkeys))
+
     return bls._AggregatePKs(list(pubkeys))
 
 
 @only_with_bls(alt_return=STUB_SIGNATURE)
 def SkToPk(SK):
-    if bls == py_ecc_bls:
-        return bls.SkToPk(SK)
+    if bls == py_ecc_bls or bls == arkworks_bls:  # no signature API in arkworks
+        return py_ecc_bls.SkToPk(SK)
     else:
         return bls.SkToPk(SK.to_bytes(32, 'big'))
 
 
 def pairing_check(values):
-    p_q_1, p_q_2 = values
-    final_exponentiation = final_exponentiate(
-        pairing(p_q_1[1], p_q_1[0], final_exponentiate=False)
-        * pairing(p_q_2[1], p_q_2[0], final_exponentiate=False)
-    )
-    return final_exponentiation == FQ12.one()
+    if bls == arkworks_bls or bls == fastest_bls:
+        p_q_1, p_q_2 = values
+        g1s = [p_q_1[0], p_q_2[0]]
+        g2s = [p_q_1[1], p_q_2[1]]
+        return arkworks_GT.multi_pairing(g1s, g2s) == arkworks_GT.one()
+    else:
+        p_q_1, p_q_2 = values
+        final_exponentiation = py_ecc_final_exponentiate(
+            py_ecc_pairing(p_q_1[1], p_q_1[0], final_exponentiate=False)
+            * py_ecc_pairing(p_q_2[1], p_q_2[0], final_exponentiate=False)
+        )
+        return final_exponentiation == py_ecc_GT.one()
+
+
+# Performs point addition of `lhs` and `rhs`
+# The points can either be in G1 or G2
+def add(lhs, rhs):
+    if bls == arkworks_bls or bls == fastest_bls:
+        return lhs + rhs
+    return py_ecc_add(lhs, rhs)
+
+
+# Performs Scalar multiplication between
+# `point` and `scalar`
+# `point` can either be in G1 or G2
+def multiply(point, scalar):
+    if bls == arkworks_bls or bls == fastest_bls:
+        int_as_bytes = scalar.to_bytes(32, 'little')
+        scalar = arkworks_Scalar.from_le_bytes(int_as_bytes)
+        return point * scalar
+    return py_ecc_mul(point, scalar)
+
+
+# Returns the point negation of `point`
+# `point` can either be in G1 or G2
+def neg(point):
+    if bls == arkworks_bls or bls == fastest_bls:
```

Comment:
I wonder if we should get rid of the `arkworks_bls` flag and just use `fastest_bls` to signify `arkworks_bls + milagro`?

Reply:
See discussion: https://github.com/ethereum/consensus-specs/pull/3272#discussion_r1118146830